### PR TITLE
Unlock php 8

### PR DESCRIPTION
**Pull request by rabauss**
Unlock php 8 + some minor fixes for php 8 - especially missing array key checks

Depends on: https://github.com/codefog/tags-bundle/pull/25

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "source": "https://github.com/terminal42/contao-node"
     },
     "require": {
-        "php": "^7.1",
+        "php": ">=7.1 || ^8.0",
```
Comment (richardhj) [inline composer.json:18]:
```suggestion
        "php": ">=7.1",
```

File: src/EventListener/DataContainerListener.php [hunk 2]
```diff
@@ -126,7 +126,7 @@ public function onPasteButtonCallback(DataContainer $dc, array $row, string $tab
         }
 
         // Disable paste into if the node is of content type
-        if (!$disablePI && NodeModel::TYPE_CONTENT === $row['type']) {
+        if (!$disablePI && isset($row['type']) && NodeModel::TYPE_CONTENT === $row['type']) {
```
Comment (richardhj) [inline src/EventListener/DataContainerListener.php:129]:
```suggestion
        if (!$disablePI && NodeModel::TYPE_CONTENT === ($row['type'] ?? '')) {
```

File: src/EventListener/DataContainerListener.php [hunk 3]
```diff
@@ -197,7 +197,7 @@ public function onEditHeaderButtonCallback(array $row, string $href, string $lab
      */
     public function onCopyButtonCallback(array $row, string $href, string $label, string $title, string $icon, string $attributes, string $table): string
     {
-        if ($GLOBALS['TL_DCA'][$table]['config']['closed']) {
+        if (isset($GLOBALS['TL_DCA'][$table]['config']['closed']) && $GLOBALS['TL_DCA'][$table]['config']['closed']) {
```
Comment (richardhj) [inline src/EventListener/DataContainerListener.php:200]:
```suggestion
        if ($GLOBALS['TL_DCA'][$table]['config']['closed'] ?? null) {
```

File: src/EventListener/DataContainerListener.php [hunk 4]
```diff
@@ -219,7 +219,7 @@ public function onCopyButtonCallback(array $row, string $href, string $label, st
      */
     public function onCopyChildsButtonCallback(array $row, string $href, string $label, string $title, string $icon, string $attributes, string $table): string
     {
-        if ($GLOBALS['TL_DCA'][$table]['config']['closed']) {
+        if (isset($GLOBALS['TL_DCA'][$table]['config']['closed']) && $GLOBALS['TL_DCA'][$table]['config']['closed']) {
```
Comment (richardhj) [inline src/EventListener/DataContainerListener.php:222]:
```suggestion
        if ($GLOBALS['TL_DCA'][$table]['config']['closed'] ?? null) {
```

**Comment by rabauss** [comment]
Thanks for reviewing - obviously after doing a lot of PRs yesterday I was confused about the allowed php version and the usage of  coalescing operator :-D

**Pull request merged**